### PR TITLE
ICU-21801 Fix array comparison warning in uspoof_impl.cpp

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21801
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by sffc** [comment]
I think this should fix the warning, although I didn't reproduce the warning locally on my version of gcc.

File: icu4c/source/i18n/uspoof_impl.cpp [hunk 1]
```diff
@@ -945,7 +945,7 @@ uspoof_swap(const UDataSwapper *ds, const void *inData, int32_t length, void *ou
     uint32_t magic = ds->readUInt32(spoofDH->fMagic);
     ds->writeUInt32((uint32_t *)&outputDH->fMagic, magic);
 
-    if (outputDH->fFormatVersion != spoofDH->fFormatVersion) {
+    if (&outputDH->fFormatVersion != &spoofDH->fFormatVersion) {
```
Comment (markusicu) [inline icu4c/source/i18n/uspoof_impl.cpp:948]: This probably works; not entirely sure without reproducing the warning.

In the ticket discussion I had proposed
```
if (inBytes != outBytes) {
```
instead, like on line 921. What do you think?

Comment (sffc) [reply icu4c/source/i18n/uspoof_impl.cpp]: sure. no strong opinion. I updated it with your suggestion.

**Review by markusicu — COMMENTED** [review]
lgtm pse squash

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**